File: Godot-MCP/server/src/llm-connectors/openrouter_client.ts
import OpenAI from 'openai'; // OpenRouter uses an OpenAI-compatible API
import { BaseLlmClient } from './base_client';

export class OpenRouterClient implements BaseLlmClient {
    private readonly client: OpenAI;
    private readonly model: string;

    constructor() {
        const apiKey = process.env.OPENROUTER_API_KEY;
        const baseUrl = process.env.OPENROUTER_BASE_URL;
        this.model = process.env.OPENROUTER_MODEL || 'anthropic/claude-3-haiku';

        if (!apiKey || apiKey.includes('YOUR_OPENROUTER_API_KEY_HERE')) {
            throw new Error('OpenRouter API key is missing or not set in .env file');
        }
        this.client = new OpenAI({ apiKey, baseURL: baseUrl });
    }

    async generatePlan(systemPrompt: string, userPrompt: string): Promise<string> {
        const response = await this.client.chat.completions.create({
            model: this.model,
            messages: [
                { role: "system", content: systemPrompt },
                { role: "user", content: userPrompt },
            ],
            response_format: { type: "json_object" },
        });
        return response.choices[0].message.content || '';
    }
}

File: bmad_method/llm_connectors/openrouter_client.py
import os
from openai import OpenAI # OpenRouter uses an OpenAI-compatible API
from .base_client import BaseLlmClient

class OpenRouterClient(BaseLlmClient):
    def __init__(self):
        super().__init__()
        self.api_key = os.getenv('OPENROUTER_API_KEY')
        self.base_url = os.getenv('OPENROUTER_BASE_URL')
        self.model = os.getenv('OPENROUTER_MODEL')
        if not self.api_key or 'YOUR_OPENROUTER_API_KEY_HERE' in self.api_key:
            raise ValueError("OpenRouter API key is missing or not set in .env file")
        self.client = OpenAI(api_key=self.api_key, base_url=self.base_url)

    def generate_plan(self, system_prompt: str, user_prompt: str) -> str:
        response = self.client.chat.completions.create(
            model=self.model,
            messages=[
                {"role": "system", "content": system_prompt},
                {"role": "user", "content": user_prompt}
            ],
            response_format={"type": "json_object"}
        )
        return response.choices[0].message.content
